interface to GAP's group recognition (first attempt)

ThomasBreuer opened the pull request:
This code provided essentially a new type, the `show` methods derived from GAP's `ViewObj` method for its `IsRecogNode` objects, and a few wrappers for functions dealing with the objects.
Let us discuss this basic setup, and how we want to proceed.

ThomasBreuer committed:
interface to GAP's group recognition (first attempt)

fingolfin commented on src/Groups/recog.jl:
```diff
@@ -0,0 +1,118 @@
+GapObj(node::GroupRecognitionNode) = node.gap_node
+
+# did the recognition not fail?
+is_ready(node::GroupRecognitionNode) = GAP.Globals.IsReady(node.gap_node)
```
```suggestion
is_ready(node::GroupRecognitionNode) = GAP.Globals.IsReady(node.gap_node)::Bool
```

fingolfin commented on src/Groups/recog.jl:
```diff
@@ -0,0 +1,118 @@
+is_ready(node::GroupRecognitionNode) = GAP.Globals.IsReady(node.gap_node)
+
+# does the node describe a leaf?
+is_leaf(node::GroupRecognitionNode) = GAP.Globals.IsLeaf(node.gap_node)
```
```suggestion
is_leaf(node::GroupRecognitionNode) = GAP.Globals.IsLeaf(node.gap_node)::Bool
```

fingolfin commented on src/Groups/recog.jl:
```diff
@@ -0,0 +1,118 @@
+
+function Base.show(io::IO, node::GroupRecognitionNode)
+  if is_terse(io)
+    print(io, "Recognition node of a group")
```
Maybe "Recognition tree" would be more apt and less confusing for users? (I've contemplate changing this name in `recog` itself but since there it'd require changing a lot of code, I've put it off for now)

fingolfin reviewed (approved):
That seems like a great start.

There is one function missing:  in GAP a group recognition object allows doing `g in recogNode` which solves the membership problem for the given group element `g`.

(Of course what we'd also want is functions for computing a word/SLP representing `g` in terms of some generators of `G`, but unfortunately this functionality is not complete in `recog` -- @danielrademacher implemented the relevant bits for the leaf cases he solved in his PhD, but we now also need to work on the general infrastructure to get it. We'll work on that, but it won't be there for the Begehung)

ThomasBreuer committed:
next iteration

- name changes as suggested in the comments
- use `GAPWrap` where possible
- change the `show` method (no newline in the end, fewer linebreaks)
- do not call `_oscar_group`
- add `Base.in` method
- add `straight_line_program` for recognition trees (and for GAP slps)
- add a few tests
- add initial documentation of group recognition

ThomasBreuer committed:
switch off GAP's info messages late enough

Switch off info messages *after* (installing and) loading GAP packages,
since packages may define new info classes which have by default
a nonzero info level.
(This happens for the Recog package.)

fingolfin reviewed (approved):
Looks good to me. Shall we just merge it now so that work on a notebook can start? We can still refine it afterwards.
